### PR TITLE
Optimized kernel config for better general kernel support and saner defaults

### PR DESCRIPTION
**Pull request by WSLUser**
This configuration was originally based on Clear Linux's kernel config options toned down for general kernel usage and enabled a number of scenarios some of which that are addressed in other simpler PRs. This should make it easier to troubleshoot the Hyper-V side for bringing support to some features. USBIP has been enabled as this is a known working scenario with use of extra software installed on both Windows and WSL2. Adds ExFAT, NTFS, and IPv6 support (still needs work on the VM side). This is part of the resolution to #114. attn: @craigloewen-msft and @sashalevin. Options not supported in 5.4 should be ignored during compilation.

### CODE REVIEW

File: Microsoft/config-wsl [hunk 1]
```diff
@@ -2256,7 +3001,7 @@ CONFIG_HYPERV=y
 CONFIG_HYPERV_TIMER=y
 CONFIG_HYPERV_UTILS=y
 CONFIG_HYPERV_BALLOON=y
-CONFIG_DXGKRNL=y
+# CONFIG_DXGKRNL is not set
```
Comment (WSLUser) [inline Microsoft/config-wsl:3004]: The kernel was unable to boot with this enabled on stable WSL2 (1903 - 2004). So I disabled it by default. As it clearly works in Insiders, this should be re-enabled when releasing for those builds.

**Comment by haosmos** [comment]
Hi! 

I compiled a linux kernel based on your fork: https://github.com/WSLUser/WSL2-Linux-Kernel/tree/WSLUser-5.8.5-config and your config: https://github.com/WSLUser/WSL2-Linux-Kernel/blob/WSLUser-5.8.5-config/Microsoft/config-wsl. 

but after I specified the path to the kernel in the .wslconfig file:

```
[wsl2]
kernel=C:\\Users\\haosmos\\vmlinux
```

my wsl 2 (ubuntu-20.04) does not start — no errors are displayed, not even any command-line characters in the terminal, — just a perfectly empty terminal screen.  

I did the following steps to compile the kernel: 

1. I downloaded the sources from repo: https://github.com/WSLUser/WSL2-Linux-Kernel.git.

2. Installed packages: `sudo apt install git bc build-essential flex bison libssl-dev libelf-dev `

3. `mv wsl-config WSL2-Linux-Kernel/arch/x86/configs/wsl_defconfig`

4. `make wsl_defconfig`

After running this command, a few messages were displayed saying that the file was overwritten, but at the end of the command, the message was displayed:  

`configuration written to .config`

5. `make -j4`
 
After that the compilation process started and was completed successfully: `Kernel: arch/x86/boot/bzImage is ready (#1).`

6. `cp vmlinux /mnt/c/Users/haosmos/`

7. in the .wslconfig file I specified:  
```
[wsl2]
kernel=C:\\Users\\haosmos\\vmlinux
```

The standard kernel 4.19.128 works well.

My environment: 

windows (x64) 2004
build: 20226
wsl2 (ubuntu-20.04)

**Comment by WSLUser** [comment]
I didn't update the kernel, just the config. Also you are supposed to cp the bzimage file and rename it how you like in the directory you copied to. https://wsl.dev/wsl2-kernel-zfs/ has instructions

**Comment by nikitalita** [comment]
Have you tested these? I don't seem to be able to boot after compiling this.

**Comment by WSLUser** [comment]
*sigh*. Follow directions I linked to. I use it just fine. I will soon update to the 5.8.12 release soon so for now you may have to answer some questions. I've also cut some stuff out to try and keep config_thermal from being re-enabled locally but haven't tested those changes yet.

File: Microsoft/config-wsl [hunk 2]
```diff
@@ -1716,17 +1949,21 @@ CONFIG_DM_BUFIO=y
 CONFIG_DM_BIO_PRISON=y
 CONFIG_DM_PERSISTENT_DATA=y
 # CONFIG_DM_UNSTRIPED is not set
-# CONFIG_DM_CRYPT is not set
+CONFIG_DM_CRYPT=y
```
Comment (bitmeise) [inline Microsoft/config-wsl:1952]: ❤

Comment (nikitalita) [reply Microsoft/config-wsl:1952]: @bitmeise might wanna check this out: https://github.com/nikitalita/mount-luks-on-wsl

Comment (bitmeise) [reply Microsoft/config-wsl:1952]: thx, will have a look.

Actually, for the future I hope we don't need to jump through those loops anymore.

**Comment by chadbrewbaker** [comment]
Should https://gist.githubusercontent.com/kruxmeier/eb4becb7ba5c16192274f6fce3a47309/raw/1300aaa38916590455c87e663af9e2732f338090/fix%2520vmwp.bat be added in the Microsoft directory? Also, when compiling under Fedora33 I got a lot of thermal config questions.

**Comment by WSLUser** [comment]
I actually compile under Clear Linux but it uses the same packages as Fedora. I haven't been able to eliminate the thermal config options yet. There is a dependency re-enabling it. I already have a 5.9 config but was hoping to update the PR after fixing it. I'll probably just push it up for now in a day or two.

**Comment by WSLUser** [comment]
Likely going to need to close this PR to target the 5.10 branch. This config will now be workable just from cloning the repo if targeting the 5.10 branch (going to the Linux kernel website directly no longer needed).

**Pull request closed without merging**